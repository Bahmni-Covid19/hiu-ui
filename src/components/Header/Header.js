import React from "react";
import HeaderStyles from "./Header.style";
import MaterialButton from "@material-ui/core/Button";
import NavigationBar from "../NavigationBar/NavigationBar";

const Header = () => {
  return (
    <HeaderStyles>
      <MaterialButton className="lo">logo</MaterialButton>
      <NavigationBar />
    </HeaderStyles>
  );
};

export default Header;
