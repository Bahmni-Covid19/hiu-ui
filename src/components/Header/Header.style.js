import styled from "styled-components";

const HeaderStyles = styled.div`
  display: flex;
  align-items: center;
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 64px;
  padding: 0;
  background: #fff;
  box-shadow: 0 4px 4px rgba(0, 0, 0, 0.24);
  z-index: 50;
  .logo {
    margin-left: 24px;
    width: 80px;
    height: 40px;
    background-color: #009688;
    color: #fff;
  }
`;

export default HeaderStyles;
