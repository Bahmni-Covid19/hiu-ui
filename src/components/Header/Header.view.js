import React from 'react';
import { Redirect } from 'react-router-dom';
import Typography from '@material-ui/core/Typography';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Button from '@material-ui/core/Button';
import Link from '@material-ui/core/Link';

const Header = () => {
  const [isLoggedOut, setIsLoggedOut] = React.useState(false);

  const logout = () => {
    localStorage.removeItem('auth-token');
    setIsLoggedOut(true);
  };

  return (
    <>
      {isLoggedOut && <Redirect to="/login" />}
      <AppBar color="textPrimary">
        <Toolbar>
          <Typography variant="h6" style={{ flexGrow: 1 }}>
            <Link href={BASE_NAME}>
              LOGO
            </Link>
          </Typography>
          <Button color="primary" variant="contained" onClick={logout}>Logout</Button>
        </Toolbar>
      </AppBar>
    </>
  );
};

export default Header;
