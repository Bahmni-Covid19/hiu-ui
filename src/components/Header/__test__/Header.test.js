import React from 'react';
import { shallow } from 'enzyme';
import { Redirect } from 'react-router-dom';
import Header from '..';
import { LogoutButton } from '../Header.style';

describe('<Header />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Header />);
    localStorage.removeItem.mockReset();
  });

  describe('HTML Structure', () => {
    it('should render properly', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should not render Redirect component when isLoggedOut is false', () => {
      expect(wrapper.find(Redirect)).toHaveLength(0);
    });

    it('should render Redirect component when isLoggedOut is true', () => {
      wrapper.find(LogoutButton).simulate('click');
      expect(wrapper.find(Redirect)).toHaveLength(1);
    });
  });

  describe('Events', () => {
    describe('logout()', () => {
      it('should call removeItem() with auth-token', () => {
        wrapper.find(LogoutButton).simulate('click');
        expect(localStorage.removeItem).toHaveBeenCalledWith('auth-token');
      });

      it('should call setState with true', () => {
        const setState = jest.fn();
        jest.spyOn(React, 'useState').mockImplementation((init) => [init, setState]);
        wrapper = shallow(<Header />);

        wrapper.find(LogoutButton).simulate('click');

        expect(setState).toHaveBeenCalledWith(true);
      });
    });
  });
});
