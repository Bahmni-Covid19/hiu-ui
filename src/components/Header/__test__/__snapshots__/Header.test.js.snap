// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly  1`] = `
"<Styled(MuiBox) className=\\"makeStyles-header-1\\" boxShadow={3}>
  <a href=\\"/\\" className=\\"makeStyles-logo-2\\">
    logo
  </a>
  <WithStyles(ForwardRef(Button)) variant=\\"contained\\" color=\\"primary\\" className=\\"makeStyles-logout-3\\" onClick={[Function: logout]}>
    Logout
  </WithStyles(ForwardRef(Button))>
</Styled(MuiBox)>"
`;
